BuildaGitServer/Base/Authentication.swift

```diff
@@ -40,9 +40,17 @@ extension ProjectAuthenticator: KeychainStringSerializable {
         
         let comps = value.componentsSeparatedByString(":")
         guard comps.count >= 4 else { throw Error.withInfo("Corrupted keychain string") }
-        guard let service = GitService(rawValue: comps[0]) else {
-            throw Error.withInfo("Unsupported service: \(comps[0])")
+
+        var service: GitService
+        switch comps[0] {
+        case GitService.GitHub.hostname():
+            service = GitService.GitHub
+        case GitService.BitBucket.hostname():
+            service = GitService.BitBucket
+        default:
+            service = GitService.EnterpriseGitHub(host: comps[0])
```

Comment:
I've got a check for this now.  I don't know if it's robust enough for you, but it was the best I could come up with.